### ISSUE TITLE
coef + dummy gives a misleading variable name

### ISSUE THREAD

**Issue by adw96** [statdivlab/rigr#100]
The use of coef in a model with dummy() gives a misleading name -- flipping the implied "baseline" category. Here's a MWE:

```{r}
devtools::install_github("statdivlab/rigr")
library(rigr)
library(tidyverse)
regress("mean", fev ~ dummy(smoke), data = fev) %>% coef %>% round(2)
regress("mean", fev ~ smoke, data = fev) %>% coef %>% round(2)
```

<img width="800" alt="Screen Shot 2022-01-20 at 9 45 43 AM" src="https://user-images.githubusercontent.com/17593682/150393309-c31cd833-8bd6-451e-bc9c-2eaaf9d0d2e7.png">

This is a major issue (in my view) because it is **silent and incorrect**. 

Any ideas?

**Comment by taylorokonek** [comment]
So first I'll note that the only thing that seems to be wrong is the labeling of the output, not the code itself, which is a good sign. To see this you can check that explicitly specifying the reference group to be either "no" or "yes" in dummy() gives you the appropriate sign for the coefficient:

regress("mean", fev ~ dummy(smoke, reference = "no"), data = fev) # gives you 0.7107
regress("mean", fev ~ dummy(smoke, reference = "yes"), data = fev) # gives you -0.7107

If you look at the print statements for the regress object rather than looking at regress(...) %>% coef(), it also seems to be giving a label that at the very least isn't incorrect ( but also not particularly informative )

The issue is in the labeling for the coefficients matrix, not the augCoefficients matrix, which is also a good sign, since the latter would be more of a pain to fix I think. From what I can see by looking into the function, the problem is in reFormatReg(), called around line 566ish in regress. You could look into what it does for dummy variables, starting with line 939 in regress_utils.R. I don't think I did much with this function to update it from uwIntroStats as it wasn't well documented and it wasn't clear what it's intent was.

**Comment by adw96** [comment]
That's really helpful, thanks, @taylorokonek !! I'll try to take a look and let you know if I have any more questions for you.

**Comment by yiqunchen** [comment]
A fix has been proposed in https://github.com/statdivlab/rigr/commit/96b7b1e41c597eedfb2800fa87ae16fdf084c208. Some examples are shown below:
![Screen Shot 2022-07-07 at 10 03 41 AM](https://user-images.githubusercontent.com/8177455/177831391-9ad0a24a-947d-446a-8589-4c30790af372.png)
![Screen Shot 2022-07-07 at 10 03 56 AM](https://user-images.githubusercontent.com/8177455/177831412-58303602-9261-47f3-8bd8-4cdd204620b2.png)
![Screen Shot 2022-07-07 at 10 04 21 AM](https://user-images.githubusercontent.com/8177455/177831454-e879af9c-4c70-4e48-be24-e3a4cd9247f7.png)
![Screen Shot 2022-07-07 at 10 04 30 AM](https://user-images.githubusercontent.com/8177455/177831460-63c213d5-a9b9-4f06-b99f-c37fe3a45680.png)

Note that since this commit is entirely operating on the name labels, it doesn't really fix the issue at https://github.com/statdivlab/rigr/issues/107.

**Comment by yiqunchen** [comment]
Note that the current fix will lead to failed unit tests since the variable names will be different (and correct!) see https://github.com/statdivlab/rigr/blob/95a092198316f468d5f2e909befacfaa52f957da/tests/testthat/test_regress.R#L893-L926. 

I will work on updating these unit tests soon.

**Comment by adw96** [comment]
Wonderful, thank you @yiqunchen ! I appreciate the deep dive!

**Comment by yiqunchen** [comment]
Changes to code and unit tests merged to main branch in [e67c415](https://github.com/statdivlab/rigr/commit/e67c415ea7826ca29a4eb8580a71dc853ffc3354); closing the issue
